firefox-ios/CredentialProvider/CredentialProviderPresenter.swift

```diff
@@ -29,25 +29,37 @@ class CredentialProviderPresenter {
     }
 
     func credentialProvisionRequested(for credentialIdentity: ASPasswordCredentialIdentity) {
-        if self.profile.logins.reopenIfClosed() != nil {
-            cancel(with: .failed)
-        } else if let id = credentialIdentity.recordIdentifier {
-            profile.logins.getLogin(id: id, completionHandler: { [weak self] result in
-                switch result {
-                case .failure:
-                    self?.cancel(with: .failed)
-                case .success(let record):
-                    if let passwordCredential = record?.passwordCredential {
-                        self?.view?.extensionContext.completeRequest(
-                            withSelectedCredential: passwordCredential,
-                            completionHandler: nil
-                        )
-                    } else {
-                        self?.cancel(with: .userInteractionRequired)
+        let maxRetries = 3
+        var currentRetry = 0
+
+        func attemptProvision() {
+            if self.profile.logins.reopenIfClosed() != nil {
+                self.cancel(with: .failed)
+            } else if let id = credentialIdentity.recordIdentifier {
```

Comment:
we can move this outside the function for an early exit from `credentialProvisionRequested`